### PR TITLE
Docs/complex0

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/core/regular/ceil.hpp [hunk 1]
```diff
@@ -36,12 +36,17 @@ namespace eve
...
 //!   **Parameters**
 //!
-//!     * `x` :  [argument](@ref eve::value).
+//!     * `x` :  [real](@ref eve::value) or [complex](@ref eve::complex) argument.
 //!
 //!    **Return value**
 //!
 //!    The smallest integer not less than `x`.
 //!
+//!      The standard proposes 4 rounding modes namely: `FE_TONEAREST`, `FE_DOWNWARD`, `FE_UPWARD`,
```
Comment (jfalcou) [inline include/eve/module/core/regular/ceil.hpp:45]: this indentation is incorrect and generate bad rendering

File: include/eve/module/core/regular/conj.hpp [hunk 2]
```diff
@@ -29,17 +29,17 @@ namespace eve
...
 //!
 //!   **Parameters**
 //!
-//!     * `x` :  [argument](@ref eve::value).
+//!     * `z` :  [real](@ref eve::value) or [complex](@ref eve::complex) argument.
 //!
 //!   **Return value**
 //!
-//!     for real inputs the call reduces to identity.
+//!     Returns the conjugate. For real inputs the call reduces to identity.
```
Comment (jfalcou) [inline include/eve/module/core/regular/conj.hpp:42]: `Return the conjugate of its argument.`

File: include/eve/module/core/regular/floor.hpp [hunk 3]
```diff
@@ -36,12 +36,17 @@ namespace eve
...
 //!   **Parameters**
 //!
-//!     * `x` :  [argument](@ref eve::value).
+//!     * `x` :  [real](@ref eve::value) or [complex](@ref eve::complex) argument.
 //!
 //!    **Return value**
 //!
 //!    The largest integer not greater than `x`.
 //!
+//!      The standard proposes 4 rounding modes namely: `FE_TONEAREST`, `FE_DOWNWARD`, `FE_UPWARD`,
```
Comment (jfalcou) [inline include/eve/module/core/regular/floor.hpp:45]: Bad indenting

File: include/eve/module/core/regular/fracscale.hpp [hunk 4]
```diff
@@ -32,22 +32,22 @@ namespace eve
...
 //!    **Return value**
 //!
-//!       *  Returns the [elementwise](@ref glossary_elementwise) reduced part of the scaled input.
-//!          The number of fraction bits retained is specified by scale. By default the internal
-//!          rounding after scaling is done to nearest integer.
-//!          The call is equivalent to `a0-roundscale(a0,scale)`
+//!      Returns the [elementwise](@ref glossary_elementwise) reduced part of the scaled input.
+//!      The number of fraction bits retained is specified by scale. By default the internal
+//!      rounding after scaling is done to nearest integer.
+//!      The call is equivalent to `a0-roundscale(a0,scale)`
```
Comment (jfalcou) [inline include/eve/module/core/regular/fracscale.hpp:50]: Not sure this is where we want that written here.

File: include/eve/module/core/regular/sqr_abs.hpp [hunk 5]
```diff
@@ -29,18 +29,25 @@ namespace eve
...
-//!     * `x` :  [argument](@ref eve::value).
+//!     * `x` :  [real](@ref eve::real_value) argument.
+//!     * `z` :  [complex](@ref eve::complex) argument.
 //!
 //!    **Return value**
 //!
-//!    value containing the [elementwise](@ref glossary_elementwise)
-//!    square of `eve::abs(x)`.
+//!    1.  value containing the [elementwise](@ref glossary_elementwise)
+//!        square of `eve::abs(x)`.
```
Comment (jfalcou) [inline include/eve/module/core/regular/sqr_abs.hpp:47]: can it fit on a single line?

File: include/eve/module/core/regular/sqr_abs.hpp [hunk 6]
```diff
@@ -29,18 +29,25 @@ namespace eve
...
+//!
+//!    2.  square of the modulus of the complex argument,
```
Comment (jfalcou) [inline include/eve/module/core/regular/sqr_abs.hpp:49]: can it fit on a single line?

File: include/eve/module/core/regular/trunc.hpp [hunk 7]
```diff
@@ -36,12 +36,17 @@ namespace eve
...
-//!     * `x` :  [argument](@ref eve::value).
+//!     * `x` :  [real](@ref eve::value) or [complex](@ref eve::complex) argument.
 //!
 //!    **Return value**
 //!
 //!    The integral part of `x` with the same sign as  `x`.
 //!
+//!    For complex inputs the trunc operation is applied to both real and imaginary parts.
+//!
+//!      The standard proposes 4 rounding modes namely: `FE_TONEAREST`, `FE_DOWNWARD`, `FE_UPWARD`,
```
Comment (jfalcou) [inline include/eve/module/core/regular/trunc.hpp:47]: Bad indenting

File: include/eve/module/math/regular/cbrt.hpp [hunk 8]
```diff
@@ -36,11 +36,11 @@ namespace eve
 //!
 //! **Parameters**
 //!
-//!    `x`:   [floating real value](@ref eve::floating_real_value).
+//!     * `x` :  [real](@ref eve::value) argument.
 //!
 //! **Return value**
 //!
-//!   Returns the [elementwise](@ref glossary_elementwise) cubic root value of the input.
+//!   Returns a [elementwise](@ref glossary_elementwise) cubic root value of the input.
```
Comment (jfalcou) [inline include/eve/module/math/regular/cbrt.hpp:43]: `Return an`

File: include/eve/module/math/regular/pow.hpp [hunk 9]
```diff
@@ -30,44 +30,56 @@ namespace eve
 //!   namespace eve
 //!   {
 //!      template< eve::floating_value T, eve::floating_value U >
-//!      T pow(T x, U y) noexcept;
+//!      auto pow(T x, U y) noexcept;                                       //1
+//!
+//!      template< eve::floating_value T, eve::floating_value U >           //2
+//!      auto pow(eve::as_complex_t<T> x, U y) noexcept;
+//!
+//!      template< eve::floating_value T, eve::floating_value U >           //2
```
Comment (jfalcou) [inline include/eve/module/math/regular/pow.hpp:38]: Should be 3 and 4 not 2 and 2

Comment (jfalcou) [reply include/eve/module/math/regular/pow.hpp:38]: OR if they are the same thing, just put  //2 once

**Review by jfalcou — CHANGES_REQUESTED** [review]
Global remark:
 - `*real version*` should be `**Real version**`
 - `*complexversion*` should be `**Complex version**`

File: include/eve/module/core/regular/conj.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/core/regular/conj.hpp:42]: changed indentation does not change rendering

File: include/eve/module/core/regular/sqr_abs.hpp (hunk 5 above)
Comment (jtlap) [reply include/eve/module/core/regular/sqr_abs.hpp:47]: the rendering will put it on a single line if the window is large enough

File: include/eve/module/core/regular/sqr_abs.hpp (hunk 6 above)
Comment (jtlap) [reply include/eve/module/core/regular/sqr_abs.hpp:49]: same reply

File: include/eve/module/math/regular/pow.hpp (hunk 9 above)
Comment (jtlap) [reply include/eve/module/math/regular/pow.hpp:38]: look at the return value description

**Pull request merged**